TypeCobol/Compiler/CodeElements/Expressions/Expression.cs

```diff
@@ -14,6 +15,24 @@ public abstract class Expression : IVisitable
 
 		public ExpressionNodeType NodeType { get; private set; }
 
+        /// <summary>
+        /// Returns true if the the characteristic element of the nodes are equivalent
+        /// For a operation or relation, the element would be the operator
+        /// For a leaf, it would be the name of the variable, ...
+        /// </summary>
+        public bool EquivalentOperator(Expression other)
+        {
+            return NodeType == other.NodeType && IsEquivalentOperator(other);
+        }
+
+        protected abstract bool IsEquivalentOperator(Expression other);
```

Comment:
Why the suffix 'Operator' ? Not every expression is made of an operator. If you're looking for a verb or verbal phrase maybe `IsEquivalent` for the public method and `CheckEquivalent` for the protected method ?